### ISSUE TITLE
undefined method `inertia_share'

### ISSUE THREAD

**Issue by andregoldstein** [inertiajs/inertia-rails#87]
Hi there I'm getting an `undefined method `inertia_share'` when trying to set a flash message after a form submission

<img width="1087" alt="Screenshot 2022-08-21 at 11 35 22" src="https://user-images.githubusercontent.com/7061578/185784896-fe2c2edb-963c-48a7-a2b0-7d232a1e969b.png">

At this stage I'm just trying to figure out the best way to get a test message working

**Comment by BrandonShar** [comment]
Hey @andregoldstein we touched base on Discord, but I figured I'd answer here too so other people running into the same problem could find it. The issue here is that `inertia_share` is a class method meant to be called at the controller level and calling it within a view method like that calls it as an instance.

The typical pattern to use flash messages would be to set up the `inertia_share` at the top of your controller, but share it within a lambda so that it's not evaluated until the method renders. It would look something like this (taken from @ledermann 's [ping crm](https://github.com/ledermann/pingcrm/blob/master/app/controllers/concerns/inertia_flash.rb) :

```rails
class MyController < ApplicationController
   inertia_share flash: -> {
      {
        success: flash.notice,
        alert: flash.alert
      }
    }
```

And then set your flash variables as normal within your controller methods.

Hope that helps!
